### ISSUE TITLE
argumentStartsOnSameLineAsPreviousArgument  crashes

### ISSUE THREAD

**Issue by zpdDG4gta8XKpMCd** [microsoft/TypeScript#32487]
```
 TypeScript Server Error (3.6.0-dev.20190719) Debug Failure. Error: Debug Failure.

at Object.assertDefined (tsserver.js:2080:24)
at Object.argumentStartsOnSameLineAsPreviousArgument (tsserver.js:113383:48)
at computeIndentation (tsserver.js:112413:55)
at processChildNode (tsserver.js:112599:44)
at processChildNodes (tsserver.js:112651:48)
at tsserver.js:112543:21
at visitNodes (tsserver.js:16716:24)
at Object.forEachChild (tsserver.js:16899:21)
at processNode (tsserver.js:112540:20)
at processChildNode (tsserver.js:112600:21)
at tsserver.js:112541:21
at visitNode (tsserver.js:16711:24)
at Object.forEachChild (tsserver.js:16805:21)
at processNode (tsserver.js:112540:20)
at processChildNode (tsserver.js:112600:21)
at processChildNodes (tsserver.js:112651:48)
at tsserver.js:112543:21
at visitNodes (tsserver.js:16716:24)
at Object.forEachChild (tsserver.js:16955:24)
at processNode (tsserver.js:112540:20..
```

**Comment by zpdDG4gta8XKpMCd** [comment]
piece of code that crashes it:
```ts
const beingCareTaking = toBeingCareTakingWorkspace<
    MonitoringWorkspace, MonitoringMainWard, MonitoringFilterWard, MonitoringWorkareaKind, EffectiveDate, MonitoringTakeout
>(beingResiding, beingFilterPresenting, outOfMonitoringWorkareaKind, (workspace, takeout) => workspace.useTakeout(takeout));
```

**Comment by zpdDG4gta8XKpMCd** [comment]
this way it works:
```ts
const beingCareTaking = toBeingCareTakingWorkspace<MonitoringWorkspace, MonitoringMainWard, MonitoringFilterWard, MonitoringWorkareaKind, EffectiveDate, MonitoringTakeout>(
    beingResiding, beingFilterPresenting, outOfMonitoringWorkareaKind, (workspace, takeout) => workspace.useTakeout(takeout)
);
```

**Comment by TobiasKrahn** [comment]
Here is an even more simplistic example:
```typescript
declare class GenericObject<T> {}

const genericObject = new GenericObject<
  {}
>();
```

formatting this code produces the following error stack:
```
<syntax> TypeScript Server Error (3.7.0-dev.20190910)
Debug Failure.
Error: Debug Failure.
    at Object.assertDefined (tsserver.js:2101:24)
    at Object.argumentStartsOnSameLineAsPreviousArgument (tsserver.js:114948:48)
    at computeIndentation (tsserver.js:113978:55)
    at processChildNode (tsserver.js:114164:44)
    at processChildNodes (tsserver.js:114216:48)
    at tsserver.js:114108:21
    at visitNodes (tsserver.js:17057:24)
    at Object.forEachChild (tsserver.js:17240:21)
    at processNode (tsserver.js:114105:20)
    at processChildNode (tsserver.js:114165:21)
    at tsserver.js:114106:21
    at visitNode (tsserver.js:17052:24)
    at Object.forEachChild (tsserver.js:17146:21)
    at processNode (tsserver.js:114105:20)
    at processChildNode (tsserver.js:114165:21)
    at processChildNodes (tsserver.js:114216:48)
    at tsserver.js:114108:21
    at visitNodes (tsserver.js:17057:24)
    at Object.forEachChild (tsserver.js:17296:24)
    at processNode (tsserver.js:114105:20)
    at processChildNode (tsserver.js:114165:21)
    at tsserver.js:114106:21
    at visitNode (tsserver.js:17052:24)
    at Object.forEachChild (tsserver.js:17294:21)
    at processNode (tsserver.js:114105:20)
    at processChildNode (tsserver.js:114165:21)
    at processChildNodes (tsserver.js:114216:48)
    at tsserver.js:114108:21
    at visitNodes (tsserver.js:17057:24)
    at Object.forEachChild (tsserver.js:17289:24)
    at processNode (tsserver.js:114105:20)
    at formatSpanWorker (tsserver.js:113920:17)
    at tsserver.js:113901:208
    at Object.getFormattingScanner (tsserver.js:112677:23)
    at formatSpan (tsserver.js:113901:31)
    at Object.formatSelection (tsserver.js:113719:20)
    at Proxy.getFormattingEditsForRange (tsserver.js:126544:34)
    at IOSession.Session.getFormattingEditsForRange (tsserver.js:135374:45)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (tsserver.js:134416:61)
    at tsserver.js:135964:88
    at IOSession.Session.executeWithRequestId (tsserver.js:135955:28)
    at IOSession.Session.executeCommand (tsserver.js:135964:33)
    at IOSession.Session.onMessage (tsserver.js:135987:35)
    at Interface.<anonymous> (tsserver.js:137302:27)
    at Interface.emit (events.js:182:13)
    at Interface._onLine (readline.js:290:10)
    at Interface._normalWrite (readline.js:433:12)
    at Socket.ondata (readline.js:149:10)
    at Socket.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at Socket.Readable.push (_stream_readable.js:219:10)
    at Pipe.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Comment by YBogomolov** [comment]
It there any chance to see this issue fixed prior 3.7.0? It breaks automatic code formatting in VSCode.
